geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -110,6 +111,47 @@ private static MemberSet calculateDiff(RegionProvider regionProvider, List<Redis
     return diff;
   }
 
+  public static Set<byte[]> sinter(RegionProvider regionProvider, List<RedisKey> keys) {
+    MemberSet result = calculateInter(regionProvider, keys);
+    if (result == null) {
+      return Collections.emptySet();
+    }
+    return result;
+  }
+
+  private static MemberSet calculateInter(RegionProvider regionProvider, List<RedisKey> keys) {
+    List<RedisSet> sets = new ArrayList<>(keys.size());
+    RedisSet smallestSet = null;
+
+    for (RedisKey key : keys) {
+      RedisSet redisSet = regionProvider.getTypedRedisData(REDIS_SET, key, true);
+      if (redisSet == NULL_REDIS_SET || redisSet.scard() == 0) {
+        return null;
+      } else {
+        if (smallestSet == null || smallestSet.scard() > redisSet.scard()) {
+          smallestSet = redisSet;
+        }
```

Comment:
should you have an else here? I don't see any need to add "smallestSet" to "sets". The current code works but causes an extra set to be compared down in the next loop that compares each item in "sets" to "smallestSet".
When you find a new smallestSet you would to first add add the old smallestSet to "sets" since it is no longer smallest.